### PR TITLE
feat/refactor: add tests for external software

### PR DESCRIPTION
**Pull request by simas232**
### Main improvements in this PR:
- feat:
  - added unit tests for BLAST+, DIAMOND, HMMER, CD-HIT and MAFFT functionalities that are utilized in RAVEN. See the individual testing functions for more details in the `testing/unit_tests` directory. Also removed `testBlast` and `testDiamond` functions as they are now duplicate and therefore no longer needed.
  - `checkInstallation`: functionality check for BLAST+, DIAMOND, HMMER, CD-HIT and MAFFT is now done by calling their corresponding unit test functions
  - `getMD5Hash`: newly added function to calculate MD5 checksums for the target file

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the [development guidelines](https://github.com/SysBioChalmers/RAVEN/wiki/DevGuidelines).
- [X] Selected `devel` as a target branch
- [ ] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/RAVEN) about this PR

### CODE REVIEW

**Comment by edkerk** [comment]
Would it make sense to wrap these testing functions in [unit tests](https://se.mathworks.com/help/matlab/matlab_prog/write-function-based-unit-tests.html), similar as done in 8dd899d7efe9ae7e367ecc03de560ad38036b633?

**Comment by simas232** [comment]
It definitely would, just these functions need assertion checks for expected output. While this would likely mean that we would need to update expected output when the binaries are updated (no problem here), I wonder if it is somehow possible to force e.g. `blastp.exe... OK` printed output as required by `checkInstallation`. Perhaps one parent function would be able to tackle this, if it is not possible to get the convenient output from testing functions themselves.

**Comment by edkerk** [comment]
I suppose if a test fails because of not matching output after a binary update, then that would be a trigger to potentially update the expected output if required. So sounds good!

File: external/getBlast.m [hunk 1]
```diff
@@ -56,59 +78,88 @@
...
         EM=['blastp did not run successfully, error: ', num2str(status)];
         dispEM(EM,true);
     end
 end
-delete([tmpDB '*']);
+if ispc
+    delete([tmpDB '\tmpDB*']);
+else
+    delete([tmpDB '/tmpDB*']);
+end
```
Comment (edkerk) [inline external/getBlast.m:133]:
```suggestion
delete([tmpDB filesep 'tmpDB*']);
```

File: external/getBlast.m [hunk 2]
```diff
@@ -56,59 +78,88 @@
...
+    end    
     if status~=0
         EM=['blastp did not run successfully, error: ', num2str(status)];
         dispEM(EM,true);
     end
+    if ispc
+        delete([tmpDB '\tmpDB*']);
+    else
+        delete([tmpDB '/tmpDB*']);
+    end
```
Comment (edkerk) [inline external/getBlast.m:162]:
```suggestion
    delete([tmpDB filesep 'tmpDB*']);
```

File: external/getDiamond.m [hunk 3]
```diff
@@ -57,59 +76,81 @@
...
         EM=['DIAMOND blastp did not run successfully, error: ', num2str(status)];
         dispEM(EM,true);
     end
 end
-delete([tmpDB '*']);
+if ispc
+    delete([tmpDB '\tmpDB*']);
+else
+    delete([tmpDB '/tmpDB*']);
+end
```
Comment (edkerk) [inline external/getDiamond.m:127]:
```suggestion
delete([tmpDB filesep 'tmpDB*']);
```

File: external/getDiamond.m [hunk 4]
```diff
@@ -57,59 +76,81 @@
...
+    end
     if status~=0
         EM=['DIAMOND blastp did not run successfully, error: ', num2str(status)];
         dispEM(EM,true);
     end
+    if ispc
+        delete([tmpDB '\tmpDB*']);
+    else
+        delete([tmpDB '/tmpDB*']);
+    end
```
Comment (edkerk) [inline external/getDiamond.m:153]:
```suggestion
    delete([tmpDB filesep 'tmpDB*']);
```

File: installation/checkInstallation.m [hunk 5]
```diff
@@ -123,75 +123,37 @@ function checkInstallation()
...
 fprintf('\n*** checkInstallation complete ***\n\n');
 end
+
+function interpretResults(results)
+if results.Failed==0 && results.Incomplete==0
+    fprintf('OK\n');
+else
+    fprintf('Not OK! Download/compile the binary and rerun checkInstallation\n');
+end
+end
```
Comment (edkerk) [inline installation/checkInstallation.m:159]:
```suggestion
end

```

**Review by edkerk — CHANGES_REQUESTED** [review]
The MAFFT test fails on my PC, but it did so before as well, so most likely a problem with my installation (Windows 10, MATLAB R2021a).

The tests involve an MD5 check, does this mean that the hash should be (possibly) be updated if the binary is updated?

**Comment by simas232** [comment]
@edkerk, please use the [the very latest MAFFT version](https://mafft.cbrc.jp/alignment/software/mafft-7.490-win64-signed.zip) to do the testing on your machine. Once this PR is merged I will create another PR for external software updates.

I have not checked whether MD5 hashes change between different versions, but we will see if it does in PR involving software updates soon anyway.

**Review by edkerk — APPROVED** [review]

**Pull request merged**